hack/e2e/apiserver-update.yaml

```diff
@@ -0,0 +1,52 @@
+apiVersion: batch/v1
+kind: Job
+metadata:
+  namespace: kube-system
+  name: apiserver-updater
+spec:
+  template:
+    spec:
+      hostNetwork: true
+      nodeSelector:
+        node-role.kubernetes.io/master: ""
+      tolerations:
+      - effect: NoSchedule
+        key: node-role.kubernetes.io/master
+
+      volumes:
+      - name: manifest
+        hostPath:
+          path: /etc/kubernetes/manifests/kube-apiserver.manifest
+          type: File
+      - name: kubecfg
+        hostPath:
+          path: /etc/kubernetes/aws-iam-authenticator/kubeconfig.yaml
+          type: File
+      - name: srv
+        hostPath:
+          path: /srv/kubernetes/kube-apiserver
+          type: Directory
+
+      containers:
+      - name: apiserver-updater
+        image: busybox
+        # this shell command adds the --authentication-token-webhook-config-file flag
+        # to the apiserver manifest, which routes authentication through the webhook.
+        # it adds the relevant line after the first appearance of the string "- args;",
+        # which is a little messy--can possibly be improved by installing yq and using it (?)
```

Comment:
Long-term, it would be easier to leverage kops existing mechanism for installing aws-iam-authenticator, and then patching it for the purpose of the test, instead of maintaining hack/e2e/deploy.yaml and these apiserver files.

From https://kops.sigs.k8s.io/authentication/#turning-on-iam-authenticator-on-an-existing-cluster:

kops equivalent to apiserver-update.yaml is:
```
Edit the clusters configuration kops edit cluster ${NAME} and add the Authentication and Authorization configs to the YAML config.
Update the clusters configuration kops update cluster ${CLUSTER_NAME} --yes
Temporarily disable aws-iam-authenticator DaemonSet kubectl patch daemonset -n kube-system aws-iam-authenticator -p '{"spec": {"template": {"spec": {"nodeSelector": {"disable-aws-iam-authenticator": "true"}}}}}'
```
kops requivalent to apiserver-restart.yaml is:
```
Perform a rolling update of the masters kops rolling-update cluster ${CLUSTER_NAME} --instance-group-roles=Master --force --yes
```

Then we just patch the `image` field of DaemonSet to be the image we are testing.

 Another option would be to reuse deploy/example.yaml instead of maintaining hack/e2e/deploy.yaml , this would have the secondary benefit of testing that the excample actually works.

I am making suggestions for what we should do in future because I do think it is a little messy and fragile to be rolling our own installation/update/restart mechanism instead of leveraging the existing ones (the purpose of the test is to test changes to the authenticator itself, not so much the installation/update/restart mechanism) , but I am not blocking this PR based on this.

Reply:
I see. I remember when I was first setting this up and looked into kops authenticator as an option, but couldn't figure out a way to put in a custom image. `kubectl patch` may do the trick, though. I'll see what I can do.

Reply:
And about the restart--I apply this job in a side test. I didn't want to end up writing some messy `exec.Command` stuff inside of a test suite, so I came up with this instead.

Reply:
And just using deploy/example.yaml doesn't work because I added a few extra mappings to it to facilitate testing.